File: Solutions/2264. Largest 3-Same-Digit Number in String/2264. Largest 3-Same-Digit Number in String.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        int res=-1;
        for(int i=2;i<num.size();++i){
            if(num[i]==num[i-1] and num[i]==num[i-2]){
                res=max(res,num[i]-'0');
            }
        }
        return res==-1?"":string(3,res+'0');
    }
};

File: Solutions/2264. Largest 3-Same-Digit Number in String/2264. Largest 3-Same-Digit Number in String.go
func largestGoodInteger(num string) string {
    res:=-1
    for i:=2;i<len(num);i+=1{
        if num[i-2]==num[i] && num[i-1]==num[i]{
            res=max(res,int(num[i]-'0'))
        }
    }
    if res==-1{
        return ""
    }
    return string(res+'0')+string(res+'0')+string(res+'0')
}

func max (i,j int) int{
    if i>j{
        return i
    }
    return j
}